src/org/rascalmpl/library/lang/rascal/tests/concrete/recovery/ErrorTreeSemanticsTests.rsc

```diff
@@ -328,6 +328,16 @@ test bool testDeepMatchAmbMemo() {
     return count == 3;
 }
 
+test bool testConcreteDeepMatchAmbMemo() {
+    Amb ambTree = ambTestTree();
+
+    // Count the number of errors that is actually found by a deep match
+    int count = (0 | it + 1 | /(AmbWord)`<AmbWord w>` := ambTree, isParseError(w));
+
+    // There will only be 3 matches if deep matches are memoized, 6 if they are not.
+    return count == 3;
+}
+
```

Comment:
pls add a test where we match on the amb node itself `(0 | it + 1 | /amb(alts) <- ambTree)`, because it's a corner case.